pkgs/development/python-modules/pyobjc-core/default.nix

```diff
@@ -0,0 +1,117 @@
+# inpired by https://github.com/input-output-hk/lace/blob/0fd166666cc454171811b307050d0815452bea82/nix/lace-blockchain-services/internal/any-darwin.nix#L339
+{
+  lib,
+  pkgs,
+  stdenv,
+  buildPythonPackage,
+  setuptools,
+  fetchPypi,
+  xcodebuild,
+  cctools,
+  darwin,
+}:
+let
+  appleSDK = darwin.apple_sdk_11_0;
+  inherit (appleSDK) MacOSX-SDK objc4;
+  darwinFrameworks = appleSDK.frameworks;
+
+  apple_libffi = stdenv.mkDerivation {
+    pname = "apple-libffi";
+    inherit (MacOSX-SDK) version;
+    dontUnpack = true;
+    installPhase = ''
+      mkdir -p $out/include $out/lib
+      cp -r ${MacOSX-SDK}/usr/include/ffi $out/include/
+      cp -r ${MacOSX-SDK}/usr/lib/libffi.* $out/lib/
+    '';
+  };
+in
+
+buildPythonPackage rec {
+  pname = "pyobjc-core";
+  version = "10.3.1";
+  src = fetchPypi {
+    pname = "pyobjc_core";
+    inherit version;
+    hash = "sha256-sgSoDMwHD5qz+K9COjolpv14fiKFCNAMTDD4rFOLpyA=";
+  };
+  pyproject = true;
+  build-system = [ setuptools ];
+
+  nativeBuildInputs = [
+    xcodebuild
+    cctools
+  ];
+
+  buildInputs = [
+    objc4
+    apple_libffi
+    darwinFrameworks.Foundation
+    darwinFrameworks.GameplayKit
+    darwinFrameworks.MetalPerformanceShaders
+  ];
+
+  checkPhase = ''
+    # TODO: This library does not follow standard testing with pytest
+    # and implemented its own test runner bootstrapping unittest
+    python3 setup.py test
```

Comment:
That command is deprecated. Can we use pytest?

Reply:
I tried, it just fails with internal errors, not running any tests. In the source code testing is managed by this code and has additional bootstrapping in setup.py